great_expectations/execution_engine/split_and_sample/data_splitter.py

```diff
@@ -17,6 +22,9 @@ class DatePart(enum.Enum):
     def __eq__(self, other):
         return self.value == other.value
 
+    def __hash__(self):
+        return hash(self.value)
+
```

Comment:
Here's one instance where the style guide will yell at you - I think we need to annotate this function (and all other dunders).

Reply:
Yep! That did the trick 🙇